src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Security/AuthorizationHandlers/MakeAmendmentOfSpecificMethodologyAuthorizationHandlerTests.cs

```diff
@@ -115,36 +116,38 @@ public class PublicationRoleTests
             public async Task
                 UserWithLinkedPublicationOwnerRoleCanCreateAmendmentOfPubliclyAccessibleMethodologyOwnedByPublication()
             {
-                await GetEnumValues<PublicationRole>().ForEachAsync(async role =>
-                {
-                    var (
-                        handler,
-                        methodologyRepository,
-                        methodologyVersionRepository,
-                        publicationRoleRepository
-                        ) = CreateHandlerAndDependencies();
-
-                    methodologyVersionRepository.Setup(mock => mock.IsPubliclyAccessible(MethodologyVersion.Id))
-                        .ReturnsAsync(true);
+                await GetEnumValues<PublicationRole>()
+                    .ToAsyncEnumerable()
```

Comment:
In hindsight, I wonder how people would have felt about us renaming our own ForEachAsync() method to ForEachAwaitAsync() and replacing the internals with source.ToAsyncEnumerable().ForEachAwaitAsync(blah), rather than us having to convert things to AsyncEnumerable again and again.

Not a biggie, but it cuts down on boilerplate lines!

Potentially we could have then also put in a safety guard to ensure that it wasn't trying to be invoked on an EF Queryable query, but hey!

Reply:
Personally, I'd just write the extra `AsAsyncEnumerable` as I don't really consider a single line to be boilerplate. It provides more clarity at what we're working with and more flexibility to chain with multiple async enumerable methods. Trading that for a single line isn't worth it IMO.

Reply:
I'd accept a little boilerplate in order to use standard C# community ways of doing things, as it reduced the amount of project-specific things a new dev has to learn. Not this is much to learn in this instance.

I don't seem to have as much concern about boilerplate compared to the rest of you though. I just don't see that much of a cost to a few extra lines here or there - I'm inclined to think things like potential for bugs trump conciseness by a long way. Maybe I've not seen enough to be jaded about it yet 😄

Reply:
:+1: